### PR TITLE
Removed some unneeded test data, re-organised SQL files.

### PR DESCRIPTION
**Pull request by marcus-bcl**
Added AUTO_SERVER on port 9092, to enable local dev connections without manually starting a server.
Added flyway.clean-on-validation-error to allow us to safely modify existing SQL files and better structure our test data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,17 +108,20 @@ curl -v http://localhost:8081/health/ping --header "Authorization: Bearer $AUTH_
 
 The service includes an in-memory data based for development purposes. This is
 a lightweight but basic schema for use in initial development work but is not
-representative of the production Delius database 
+representative of the production Delius database.
 
 * H2 Web console - <http://localhost:8080/h2-console>
-* JDBC URL: `jdbc:h2:file:/tmp/hmpps-delius-api-dev;Mode=Oracle`
+* JDBC URL: `jdbc:h2:file:./dev;MODE=Oracle;AUTO_SERVER=TRUE;AUTO_SERVER_PORT=9092`
```
Comment (axle-h) [inline README.md:114]: nice

File: README.md [hunk 2]
```diff
@@ -108,17 +108,20 @@ curl -v http://localhost:8081/health/ping --header "Authorization: Bearer $AUTH_
...
 * USER: `sa`
 * PASSWORD: `password`
 
-You can also connect to H2 remotely, useful for Intellij database tools &
-using the extra code sense it adds to JPA. 
- 
-`jdbc:h2:tcp://localhost:9092/file:/tmp/hmpps-delius-api-dev;Mode=Oracle`
```
Comment (axle-h) [inline README.md:121]: what's the jdbc url now?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/deliusapi/util/Fake.kt [hunk 3]
```diff
@@ -43,13 +43,15 @@ object Fake {
   fun randomLocalDateTime(): LocalDateTime = faker.date().past(10, TimeUnit.DAYS).toLocalDateTime()
   fun randomFutureLocalDate(): LocalDate = faker.date().future(10, TimeUnit.DAYS).toLocalDate()
 
+  fun randomPositiveId(): Long = faker.number().numberBetween(1L, 900_000_000_000_000_000L) // maxvalue of db sequences
```
Comment (axle-h) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/deliusapi/util/Fake.kt:46]: Ive added the exact same thing locally, can you rename this to just `id()` and then I'll have far fewer non auto-solvable merge conflicts?

File: README.md (hunk 2 above)
Comment (marcus-bcl) [reply README.md:121]: It's the one a few lines above - `jdbc:h2:file:./dev;MODE=Oracle;AUTO_SERVER=TRUE;AUTO_SERVER_PORT=9092`, you can just paste that as-is into the URL field in the H2 connection window in IntelliJ.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/deliusapi/util/Fake.kt (hunk 3 above)
Comment (marcus-bcl) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/deliusapi/util/Fake.kt:46]: Yep will do.

**Review by axle-h — APPROVED** [review]

**Pull request merged**